cmd/snap-gdb-shim/snap-gdb-shim.c

```diff
@@ -15,20 +15,64 @@
  *
  */
 
+#include <errno.h>
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 
 #include "../libsnap-confine-private/utils.h"
 
+int must_parse_int(const char *s)
+{
+	char *endptr = NULL;
+
+	errno = 0;
+	long i = strtol(s, &endptr, 10);
+	if (errno != 0 || s == endptr || *endptr != '\0' || i < 0) {
+		die("cannot parse number in '%s'", s);
+	}
+	return i;
+}
+
 int main(int argc, char **argv)
 {
 	if (sc_is_debug_enabled()) {
 		for (int i = 0; i < argc; i++) {
 			printf("-%s-\n", argv[i]);
 		}
 	}
+
+	if (getuid() == 0) {
+		// check if we run as SUDO and if so switch to a normal user
+		const char *sudo_uid_env = getenv("SUDO_UID");
+		if (sudo_uid_env != NULL) {
+			int sudo_uid = must_parse_int(sudo_uid_env);
+			if (sudo_uid != 0) {
+				if (setuid(sudo_uid) != 0) {
```

Comment:
Nitpick: C code in snapd has a style of `< 0` comparisons, unless the used function returns a non-standard error indicator.